File: daily-ask/infrastructure/config/configure-environment-variables.go
package config

import (
	"fmt"
	"log"
	"os"
	"path"

	"github.com/joho/godotenv"
)

type EnvironmentVariables struct {
	DatabaseUrl      string
	DatabaseName     string
	DatabaseUsername string
	DatabasePassword string
	MessageBrokerUrl string
}

func getEnvironmentVariable(name string) string {
	environmentVariable := os.Getenv(name)
	if environmentVariable == "" {
		log.Panicf("%s is not defined", name)
	}
	return environmentVariable
}

func ConfigureEnvironmentVariables() (*EnvironmentVariables, error) {
	dir, err := os.Getwd()
	if err != nil {
		return nil, err
	}

	err = godotenv.Load(path.Join(dir, ".env"))
	if err != nil {
		return nil, fmt.Errorf("failed to load .env file: %s", err.Error())
	}

	databaseName := getEnvironmentVariable("DATABASE_NAME")
	databaseHost := getEnvironmentVariable("DATABASE_HOST")
	databasePort := getEnvironmentVariable("DATABASE_PORT")
	databaseUsername := getEnvironmentVariable("DATABASE_USERNAME")
	databasePassword := getEnvironmentVariable("DATABASE_PASSWORD")
	messageBrokerHost := getEnvironmentVariable("MESSAGE_BROKER_HOST")
	messageBrokerPort := getEnvironmentVariable("MESSAGE_BROKER_PORT")

	databaseUrl := fmt.Sprintf("%s:%s", databaseHost, databasePort)
	messageBrokerUrl := fmt.Sprintf("%s:%s", messageBrokerHost, messageBrokerPort)

	return &EnvironmentVariables{
		DatabaseUrl:      databaseUrl,
		DatabaseName:     databaseName,
		DatabaseUsername: databaseUsername,
		DatabasePassword: databasePassword,
		MessageBrokerUrl: messageBrokerUrl,
	}, nil
}

File: daily-ask/src/infrastructure/config/configure-environment-variables.ts
import dotenv from 'dotenv';

const configureEnvironmentVariables = () => {
  dotenv.config();

  const {
    MESSAGE_BROKER_HOST,
    MESSAGE_BROKER_PORT,
  } = process.env;

  if (!MESSAGE_BROKER_HOST) {
    throw new Error('MESSAGE_BROKER_HOST is not defined.');
  }
  if (!MESSAGE_BROKER_PORT) {
    throw new Error('MESSAGE_BROKER_PORT is not defined.');
  }

  const MESSAGE_BROKER_URL = `${MESSAGE_BROKER_HOST}:${MESSAGE_BROKER_PORT}`;

  return { MESSAGE_BROKER_URL };
};

export default configureEnvironmentVariables;
